cyipopt/tests/unit/test_scipy_ipopt_from_scipy.py

```diff
@@ -1125,7 +1125,13 @@ def nci(x):
     ref = minimize(fun=obj, x0=x0, method='slsqp',
                    bounds=bnds, constraints=nlcs)
     assert ref.success
-    assert res.fun <= ref.fun
+    assert res.fun <= ref.fun  # Ipopt legitimately does better than slsqp here
+
+    # If we give SLSQP a good guess, it agrees with Ipopt
+    ref2 = minimize(fun=obj, x0=res.x, method='slsqp',
+                   bounds=bnds, constraints=nlcs)
+    assert ref2.success
+    assert_allclose(ref2.fun, res.fun)
```

Comment:
_Note: this change is unrelated to the rest of the PR, but since I needed to re-run CI, I decided to do it here rather than opening another PR._

While developing gh-207, I noticed two things about `minimize_ipopt` that were concerning. One is addressed by gh-229; this addresses the other. I wasn't sure before, but I see now that Ipopt really does find a better solution to this problem than SLSQP. This extension confirms that the solution found by Ipopt is legitimate because SLSQP agrees when the solution is provided as a guess.